xchem_db/models.py

```diff
@@ -102,11 +268,122 @@ class Crystal(models.Model):
 
     status = models.CharField(choices=CHOICES, max_length=2, default=PREPROCESSING)
 
+    #added SoakDB attributes
+
+    crystal_plate = models.ForeignKey(CrystalPlate, blank=True, null=True, on_delete=models.PROTECT) #need to do something about nullability
+    well = models.CharField(max_length=4,  blank=True, null=True)
+    echo_x = models.IntegerField(blank=True, null=True,) #double-check if it shouldn't be float
+    echo_y = models.IntegerField(blank=True, null=True,) #double-check if it shouldn't be float
+    score = models.IntegerField(blank=True, null=True,)
+#    image -- need to figure out how this will be stored
+
     class Meta:
         if os.getcwd() != '/dls/science/groups/i04-1/software/luigi_pipeline/pipelineDEV':
             app_label = 'xchem_db'
         db_table = 'crystal'
-        unique_together = ('crystal_name', 'visit', 'compound', 'product')
+#        unique_together = ('crystal_name', 'visit', 'compound', 'product')
+        unique_together = ('crystal_name', 'visit', 'product') #removed compound from unique_together to allow for cocktails
+
+#new class
+class SolventNotes(models.Model):
+    '''To store user's notes on conclusions from solvent testing; values
+    not to be processed any further except for reminding the user to 
+    apply cryo'''
+    
+    proposal = models.ForeignKey(Proposals, on_delete=models.CASCADE)
+    solvent = models.CharField(max_length=32, blank=True, null=True)
+    solvent_concentration = models.FloatField(blank=True, null=True)
+    soak_time = models.DurationField(blank=True, null=True)
+    cryo = models.CharField(max_length=32, blank=True, null=True)
+    cryo_concentration = models.FloatField(blank=True, null=True)
+    comments = models.TextField(blank=True, null=True)
+    
+    class Meta:
+        if os.getcwd() != '/dls/science/groups/i04-1/software/luigi_pipeline/pipelineDEV':
+            app_label = 'xchem_db'
+        db_table = 'solvent_notes'
+
+
+#new class (SoakDB experimental data)	
+class SoakDBCompound(models.Model):
+    '''Compound data copied from inventory data when the compound is used
+    in the experiment'''
+
+    proposal = models.ForeignKey(Proposals, related_name="exp_compounds", on_delete=models.CASCADE, blank=True, null=True)
+    library_name = models.CharField(max_length=100)
+    library_plate = models.CharField(max_length=100)
+    well = models.CharField(max_length=4)
+    code = models.CharField(max_length=100)
+    smiles = models.CharField(max_length=256)
+    crystal = models.ForeignKey(Crystal, related_name="compounds", on_delete=models.PROTECT, blank=True, null=True) #to allow cocktails
+    
+    class Meta:
+        if os.getcwd() != '/dls/science/groups/i04-1/software/luigi_pipeline/pipelineDEV':
+            app_label = 'xchem_db'
+        db_table = 'soakdb_compound'
+
```

Comment:
We 'XChem' need to have an extremely low-level Frank-less discussion about how we should implement combi soak crystals.